code/modules/reagents/chemistry/holder.dm

```diff
@@ -476,6 +504,25 @@
 		del_reagent(R.id)
 	return 0
 
+/datum/reagents/proc/reaction_check(mob/living/M, datum/reagent/R)
```

Comment:
As nice as this proc is, it is never actually invoked anywhere. Please make it be used?

Reply:
Now added it in `proc/reaction`, like 538.